Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteDatabase.mm

```diff
@@ -0,0 +1,243 @@
+        return result;
+    }
+
+    _handle = nullptr;
+    return result;
+}
+
+- (BOOL)reportErrorWithCode:(_WKSQLiteErrorCode)errorCode query:(NSString *)query error:(NSError **)error
+{
+    dispatch_assert_queue(_queue);
+    ASSERT(errorCode != SQLITE_OK);
+
+    static const int expectedDictionarySize = 3;
+
+    NSMutableDictionary *userInfo = [NSMutableDictionary dictionaryWithCapacity:expectedDictionarySize];
+
+    if (_url)
+        userInfo[NSURLErrorKey] = [_url copy];
+
+    if (_WKSQLiteErrorCode(errorCode)) {
+        if (const char* errMsg = sqlite3_errmsg(_handle))
+            userInfo[_WKWebExtensionSQLiteErrorMessageKey] = @(errMsg);
+    }
+
+    if (query)
+        userInfo[_WKWebExtensionSQLiteErrorSQLKey] = [query copy];
+
+    NSError *errorObject = [NSError errorWithDomain:_WKWebExtensionSQLiteErrorDomain code:errorCode userInfo:userInfo];
+
+    if (error)
+        *error = errorObject;
+    else
+        RELEASE_LOG_ERROR(Extensions, "Unhandled SQLite error: %{public}@", privacyPreservingDescription(errorObject));
+
+    return NO;
+}
+
+- (BOOL)reportErrorWithCode:(_WKSQLiteErrorCode)errorCode statement:(sqlite3_stmt*)statement error:(NSError **)error
+{
+    dispatch_assert_queue(_queue);
+    ASSERT(errorCode != SQLITE_OK);
+
+    if (statement) {
+        if (char* sql = sqlite3_expanded_sql(statement)) {
+            BOOL result = [self reportErrorWithCode:errorCode query:@(sql) error:error];
+            sqlite3_free(sql);
+            return result;
+        }
+    }
+
+    return [self reportErrorWithCode:errorCode query:nil error:error];
+}
+
+- (BOOL)enableWAL:(NSError **)error
+{
+    return SQLiteDatabaseEnumerate(self, error, @"PRAGMA journal_mode = WAL", std::tie(std::ignore));
+    return SQLiteDatabaseEnumerate(self, error, @"PRAGMA journal_mode = WAL", std::tie(std::ignore));
+}
+
+
+- (BOOL)openWithAccessType:(SQLiteDatabaseAccessType)accessType error:(NSError **)error
+{
+    return [self openWithAccessType:accessType vfs:nil error:error];
+}
+
+- (BOOL)openWithAccessType:(SQLiteDatabaseAccessType)accessType vfs:(NSString *)vfs error:(NSError **)error
+{
+    return [self openWithAccessType:accessType protectionType:SQLiteDatabaseProtectionTypeDefault vfs:vfs error:error];
+}
+
+- (BOOL)openWithAccessType:(SQLiteDatabaseAccessType)accessType protectionType:(SQLiteDatabaseProtectionType)protectionType vfs:(NSString *)vfs error:(NSError **)error
+{
+    int flags = SQLITE_OPEN_NOMUTEX;
+
+    switch (accessType) {
+    case SQLiteDatabaseAccessTypeReadOnly:
+        flags |= SQLITE_OPEN_READONLY;
+        break;
+    case SQLiteDatabaseAccessTypeReadWrite:
+        flags |= SQLITE_OPEN_READWRITE;
+        break;
+    case SQLiteDatabaseAccessTypeReadWriteCreate:
+        flags |= SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE;
+        break;
+    }
+
+#if PLATFORM(IOS_FAMILY)
+    switch (protectionType) {
+    case SQLiteDatabaseProtectionType:
```

Comment:
I think this line is the build failure on iOS, it can just be removed, I think.

Reply:
was supposed to be `SQLiteDatabaseProtectionTypeDefault`. will update